src/input_device/JoystickInput.cpp

```diff
@@ -76,27 +82,28 @@ PlayerInputAbs JoystickInputDevice::transferInput()
 
 bool JoystickInputDevice::getAction(const JoystickAction& action)
 {
-	if (action.joy != nullptr)
+	SDL_Joystick* joystick = mInputManager->getJoystickPool().getJoystick(action.joyid);

```

Comment:
should `InputManager` get a `getJoystick` convenience function? Then JoystickInput would not even need to know about the existence of the JoystickPool class, I think,

Reply:
This is a much better interface. I change it.